ohapi/api.py

```diff
@@ -173,3 +173,44 @@ def delete_file(access_token, project_member_id, base_url=OH_BASE_URL,
 # Alternate names for the same functions.
 def delete_files(*args, **kwargs):
     return delete_file(*args, **kwargs)
+
+
+def send_message_individual(subject, message, access_token, base_url=OH_BASE_URL):
+    """ 
+    send message to individual member.
+    """
+    url = urlparse.urljoin(
+        base_url, '/api/direct-sharing/project/message/?{}'.format(
+        urlparse.urlencode({'access_token': access_token})))
+    requests.post(url,data={'subject': subject,
+                            'message': message})
+
+
+def send_message_multiple(subject, message, master_access_token, all_members, project_member_ids, base_url=OH_BASE_URL):
+    """
+    send message to multiple members.
+    """
+    url = urlparse.urljoin(
+        base_url, '/api/direct-sharing/project/message/?{}'.format(
+        urlparse.urlencode({'access_token': master_access_token})))
+    if all_members  and project_member_ids:
+        raise ValueError(
+            "One (and only one) of the following must be specified: "
+            "project_members_id or all_members is set to True.")
+    requests.post(url, data={'all_members': all_members,
+                 'project_member_ids': project_member_ids,
+                 'subject': subject,
+                 'message': message})
+
+
+def messaging(subject, message, token, is_token_master=False, all_members=False, project_member_ids=None, base_url=OH_BASE_URL):
```

Comment:
I'd like this to be renamed to `message`, and the parameter `token` to change to `access_token` to be consistent with the other code.

I would rather not have a parameter called `is_token_master`. I want the function to mirror the API's own interface – take the parameters the API wants, and no more than that, and pass them to the API. I don't think separate functions for individual and multiple are necessary. If project_member_ids isn't provided, don't provide it to the API.

(The API might produce an error, e.g. if an OAuth2 token is used in conjunction with `all_members=True`. I think we could do a better job of reporting API errors in this package. But that's a general issue, let's not worry about it here.)

One thing you might add is an error if project_member_ids and all_members are both provided, since we know that will result in an error in the API.

Reply:
(and as a postscript, I can see that some of what I disagree with was actually suggested by @gedankenstuecke. I apologize for the confusion, but I want `api.py` to be a minimalist wrapper and essentially mirror the API's own parameters for each API endpoint.)